CMakeLists.txt

```diff
@@ -3,48 +3,99 @@ cmake_minimum_required( VERSION 3.0 FATAL_ERROR )
 project( libmygpo-qt )
 
 set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
-set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII" )
 
-option(BUILD_WITH_QT4 "Build libmygpo-qt with Qt4" OFF)
-option(MYGPO_BUILD_TESTS "Build all unit tests" ON)
+include(CheckCXXCompilerFlag)
+
+list(APPEND COMPILE_OPTIONS
+  -Wall
+  -Wextra
+  -Wpedantic
+  -Wunused
+  -Wshadow
+  -Wundef
+  -Wpacked
+  -Wuninitialized
+  -Wredundant-decls
+  -Wunused-parameter
+  -Wcast-align
+  -Wchar-subscripts
+  -Wpointer-arith
+  -Wwrite-strings
+  -Wformat-security
+  -Wold-style-cast
+  -Woverloaded-virtual
+  -Wnon-virtual-dtor
+  -Wstack-protector
+  -Winit-self
+  -Wmissing-format-attribute
+  -Wmissing-include-dirs
+  -Wmissing-declarations
+  -Wstrict-overflow=2
+  -Wformat=2
+  -Wdisabled-optimization
+  -fno-inline
+)
+if ( NOT WIN32 )
+  list(APPEND COMPILE_OPTIONS -fvisibility=hidden )
+endif()
+
+if (CMAKE_COMPILER_IS_GNUCXX)
+  list(APPEND COMPILE_OPTIONS
+    -ggdb3
+    -Wunsafe-loop-optimizations
+  )
+  check_cxx_compiler_flag( -Wlogical-op GNUCXX_HAS_WLOGICAL_OP )
+  if ( GNUCXX_HAS_WLOGICAL_OP )
+    list(APPEND COMPILE_OPTIONS -Wlogical-op )
+  endif()
+endif (CMAKE_COMPILER_IS_GNUCXX)
+
+add_compile_options(${COMPILE_OPTIONS})
+
+add_definitions(-DQT_STRICT_ITERATOR -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII)
+
+option( MYGPO_QT_MAJOR_VERSION "Qt version to use (5 or 6)" )
+option( BUILD_WITH_QT5 "Build libmygpo-qt with Qt 5" OFF )
+option( BUILD_WITH_QT6 "Build libmygpo-qt with Qt 6" OFF )
+option( MYGPO_BUILD_TESTS "Build all unit tests" ON )
+
+set(QT_COMPONENTS Core Network)
+if(MYGPO_BUILD_TESTS)
+    list(APPEND QT_COMPONENTS Test)
+endif()
 
-if( NOT BUILD_WITH_QT4 )
-    if( MYGPO_BUILD_TESTS )
-        find_package(Qt5 REQUIRED COMPONENTS Core Network Test CONFIG)
+if( BUILD_WITH_QT6 OR MYGPO_QT_MAJOR_VERSION EQUAL 6 )
+    set( MYGPO_QT_MAJOR_VERSION 6 CACHE STRING "" FORCE )
+    set( BUILD_WITH_QT6 ON CACHE BOOL "" FORCE )
+elseif( BUILD_WITH_QT5 OR MYGPO_QT_MAJOR_VERSION EQUAL 5 )
+    set( MYGPO_QT_MAJOR_VERSION 5 CACHE STRING "" FORCE )
+    set( BUILD_WITH_QT5 ON CACHE BOOL "" FORCE )
+    set( QT_MIN_VERSION 5.2 )
+ else()
+    # Automatically detect Qt version.
+    find_package( QT NAMES Qt6 Qt5 COMPONENTS ${QT_COMPONENTS} REQUIRED )
+    if( QT_FOUND AND QT_VERSION_MAJOR EQUAL 6 )
+        set( BUILD_WITH_QT6 ON CACHE BOOL "" FORCE )
+        set( MYGPO_QT_MAJOR_VERSION 6 CACHE STRING "" FORCE )
+    elseif (QT_FOUND AND QT_VERSION_MAJOR EQUAL 5 )
+        set( BUILD_WITH_QT5 ON CACHE BOOL "" FORCE )
+        set( MYGPO_QT_MAJOR_VERSION 5 CACHE STRING "" FORCE )
+        set( QT_MIN_VERSION 5.2 )
     else()
-        find_package(Qt5 REQUIRED COMPONENTS Core Network CONFIG)
+        message( FATAL_ERROR "Qt 5 and 6 was not found." )
```

Comment:
The message makes it sound like both are needed.

Suggestions:
- "Neither Qt 5 nor 6 were found."
- "Could not find a suitable Qt version."